provider/fs/src/deserializer.rs

```diff
@@ -3,30 +3,50 @@
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
 use crate::manifest::SyntaxOption;
+use displaydoc::Display;
 use icu_provider::prelude::*;
 use icu_provider::serde::*;
 use icu_provider::yoke::trait_hack::YokeTraitHack;
 use icu_provider::yoke::Yokeable;
 use serde::Deserialize;
 use std::path::Path;
 use std::rc::Rc;
-use thiserror::Error;
 
 /// An Error type specifically for the [`Deserializer`](serde::Deserializer) that doesn't carry filenames
-#[derive(Error, Debug)]
+#[derive(Display, Debug)]
 pub enum Error {
-    #[error(transparent)]
-    Json(#[from] serde_json::error::Error),
+    #[displaydoc("{0}")]
+    Json(serde_json::error::Error),
     #[cfg(feature = "bincode")]
-    #[error(transparent)]
-    Bincode(#[from] bincode::Error),
-    #[error(transparent)]
-    DataProvider(#[from] DataError),
+    #[displaydoc("{0}")]
+    Bincode(bincode::Error),
+    #[displaydoc("{0}")]
+    DataProvider(DataError),
     #[allow(dead_code)]
-    #[error("Unknown syntax: {0:?}")]
+    #[displaydoc("Unknown syntax: {0:?}")]
     UnknownSyntax(SyntaxOption),
 }
 
+impl std::error::Error for Error {}
+
+impl From<serde_json::error::Error> for Error {
```

Comment:
Question: do we have a path forward for getting rid of these `From` impls again in the future?

Reply:
It's easy to write another custom derive for `From` for stuff like this.

But also, manually writing From impls like this is super common.